// @builder.io/sdk-vue/lib/node/repeated-block-zjZTsi5s.js
import { defineComponent as t, defineAsyncComponent as n, resolveComponent as r, openBlock as c, createBlock as p } from "vue";
import { i as s, d as l } from "./index-6f7aS4as.js";
const i = () => import("./index-6f7aS4as.js").then((o) => o.D).then((o) => o.default).catch((o) => {
  throw console.error(
    "Error while attempting to dynamically import component Block at ../block.vue",
    o
  ), o;
}), m = t({
  name: "repeated-block",
  components: { Block: n(i) },
  props: ["repeatContext", "block", "registeredComponents", "linkComponent"],
  data() {
    return { store: this.repeatContext };
  },
  provide() {
    const o = this;
    return {
      [s.key]: o.store
    };
  }
});
function a(o, k, d, C, f, u) {
  const e = r("Block");
  return c(), p(e, {
    block: o.block,
    context: o.store,
    registeredComponents: o.registeredComponents,
    linkComponent: o.linkComponent
  }, null, 8, ["block", "context", "registeredComponents", "linkComponent"]);
}
const _ = /* @__PURE__ */ l(m, [["render", a]]);
export {
  _ as default
};
