// @builder.io/sdk-vue/lib/edge/repeated-block-730A9lRm.js
import { defineComponent as t, defineAsyncComponent as n, resolveComponent as r, openBlock as c, createBlock as p } from "vue";
import { j as s, e as l } from "./index-O2Rm1-0p.js";
const i = () => import("./index-O2Rm1-0p.js").then((e) => e.H).then((e) => e.default).catch((e) => {
  throw console.error(
    "Error while attempting to dynamically import component Block at ../block.vue",
    e
  ), e;
}), m = t({
  name: "repeated-block",
  components: { Block: n(i) },
  props: ["repeatContext", "block", "registeredComponents", "linkComponent"],
  data() {
    return { store: this.repeatContext };
  },
  provide() {
    const e = this;
    return {
      [s.key]: e.store
    };
  }
});
function a(e, k, d, C, f, u) {
  const o = r("Block");
  return c(), p(o, {
    block: e.block,
    context: e.store,
    registeredComponents: e.registeredComponents,
    linkComponent: e.linkComponent
  }, null, 8, ["block", "context", "registeredComponents", "linkComponent"]);
}
const _ = /* @__PURE__ */ l(m, [["render", a]]);
export {
  _ as default
};
